packages/form-layout/src/vaadin-form-item.js

```diff
@@ -164,14 +165,72 @@ class FormItem extends ThemableMixin(PolymerElement) {
     super();
     this.__updateInvalidState = this.__updateInvalidState.bind(this);
     this.__contentFieldObserver = new MutationObserver(() => this.__updateRequiredState(this.__contentField.required));
+
+    // Ensure every instance has unique ID
+    const uniqueId = (FormItem._uniqueLabelId = 1 + FormItem._uniqueLabelId || 0);
+    this.__labelId = `label-${this.localName}-${uniqueId}`;
+  }
+
+  /**
+   * An array of field nodes of the content slot.
+   *
+   * @return {HTMLElement[]}
+   * @private
+   */
+  get __fieldNodes() {
+    return this.$.contentSlot.assignedElements();
+  }
+
+  /**
+   * The first field node in the content slot.
+   *
+   * @return {HTMLElement | undefined}
+   * @private
+   */
+  get __fieldNode() {
+    return this.__fieldNodes[0];
+  }
+
+  /**
+   * The first label node in the label slot.
+   *
+   * @type {HTMLElement | undefined}
+   * @private
+   */
+  get __labelNode() {
+    return this.$.labelSlot.assignedElements()[0];
+  }
+
+  /**
+   * Links the label to a field by adding the label's id to
+   * the `aria-labelledby` attribute of the field.
+   *
+   * @param {HTMLElement} field
+   * @private
+   */
+  __linkLabelToField(field) {
+    const ariaTarget = field.ariaTarget || field;
```

Comment:
1. Let's move this to a protected method e.g. `_getFieldTarget(field)` to make overriding possible.
2. Let's briefly document this logic in the `vaadin-form-item` JSDoc. We should at least mention that `ariaTarget` is a special property used by Vaadin field components that comes from `FieldMixin`. Let's also note that the element used as `ariaTarget` has to be in the light DOM (not in Shadow DOM).

Reply:
Decided internally not to mention the `ariaTarget` property directly in the component's JSDoc description since it is quite a technical detail.

Instead, I described it in the JSDoc of the `getFieldTarget(field)` method.